coffea/hist/__init__.py

```diff
@@ -17,9 +17,6 @@
     Cat,
     StringBin,
 )
-from .export import (
```

Comment:
Mostly I didn't want the awkward0 deprecation warning showing up any time anyone imports hist.
I can't seem to turn of the warning being emitted.

Since it inherits from exception I guess I can catch it or something? I'll poke at it.

Reply:
Can you filter it with the [warnings filter](https://docs.python.org/3/library/warnings.html#describing-warning-filters)? It's a `FutureWarning`.

Reply:
Yes, you can do this:

```python
>>> import warnings
>>> with warnings.catch_warnings():
...     warnings.simplefilter("ignore")
...     import awkward0
... 
>>> awkward0
<module 'awkward0' from '.../python3.8/site-packages/awkward0/__init__.py'>
```

Reply:
OK - so long as that doesn't set the global warning filter state and a subsequent import of awkward0 into the global namespace would cause a warning.

Reply:
This sets the warning state only temporarily—after the `with` block, `FutureWarning` would be shown again. However, once `awkward0` is loaded, `import awkward0` is a pass-through (just because of how Python's import works), so users will never see the Awkward0 warning.

Reply:
Ah, hmm. I think I'll take it since we're removing awkward0 support from the rest of coffea and we shouldn't have major issues from version mixups.

Reply:
This warning-hiding might be relevant for importing Uproot 3, which is still needed for a couple of things. Or just leave the warnings in and explain them when they come up, or catch Uproot 3's warning and emit your own, saying, "We rely on Uproot 3 for this and that, but there's no way around that for now."

The thing a user needs to know is whether an action needs to be taken. If they're manually importing Uproot 3, the answer is often "yes" (they should upgrade if they are just starting a project), but if they're getting it as a dependency through Coffea, the answer is "no" (Coffea will upgrade underneath when it becomes possible; the user has no action to take).

Reply:
Can we hide the `TH1` class declaration that requires importing uproot3 inside the export1d function so that uproot3 is only imported (and awkward0 along with) when the function is called?

Reply:
That makes the TH1 class unpickleable (we actually have a test for this, contributed by the Aachen group).

Reply:
I think I'll just go with wrapping the import, since in time this will switch to uproot4.